src/sentry/api/endpoints/release_thresholds/utils.py

```diff
@@ -0,0 +1,54 @@
+from typing import Any, Dict, List
+
+from snuba_sdk import Request as SnubaRequest
+from snuba_sdk.column import Column
+from snuba_sdk.conditions import Condition, Op
+from snuba_sdk.entity import Entity
+from snuba_sdk.expressions import Granularity
+from snuba_sdk.function import Function
+from snuba_sdk.orderby import Direction, OrderBy
+from snuba_sdk.query import Query
+
+from sentry.snuba.dataset import Dataset
+from sentry.utils import snuba
+
+
+def get_errors_timeseries_counts_by_project_and_release(
```

Comment:
So I noticed there was a discussion on what utils to use, but I just wanted to add that I think we could accomplish with the `timeseries_query` util @wmak mentioned by doing:

```python
results = timeseries_query(
    selected_columns=['count()'],
    query=f"event.type:error release:[{releases}]",
    params=self.get_snuba_params(request, organization),
    rollup=60,
    referrer="snuba.sessions.check_releases_have_health_data"
```

`get_snuba_params` handles stuff with the time range options, allowing for absolute timestamps or ranges such as "30d" as well as the project and environment filters. Will your usage potentially require more filtering defined by the user?

Reply:
Nope. it will not.

I tried looking into the timeseries_query stuff... TBH it seems overly complicated and far too overengineered...
I just need a simple way to fetch error count timeseries. It's way too difficult to understand what's going on with the `get_snuba_params` and `timeseries_query`, and at this point i'm not willing to bother spending the effort to decipher them :/

Reply:
I'd highly recommend using Nar's suggestion here. I'm not sure why you need to decipher anything when he's already given you the code to copy and paste.

But if you want to ignore our suggestions that's up to you 🤷

Reply:
Ah i see what you mean about the time stuff...
`statsPeriod`. I just posted in slack about these.
I think `statsPeriod` is just bad code and we should get rid of them. I definitely don't want to be forced to utilize this complicated tightly coupled code _just_ for that 😬 

sorry if i'm being too blunt here haha

Reply:
🤷 Depends if you need to get your project finished or if you have the free time to redesign the entire stats period API.

Projects at Sentry's size will always have 9001 things that you can fix, but it's always vital to prioritize what's important first.

Reply:
👍 understood

To close out on this discussion - there's a few reasons why i'm choosing not to utilize `get_snuba_params` nor `timeseries_query`

- I would actually prefer _not_ to allow customers to manipulate the snuba query directly from their request params. This is an implementation smell i think could be fairly dangerous. 
here i'll be explicitly requiring the filter params to fetch the full set of events. Eventually, i'll be writing those results to the DB as well, so i don't want to introduce any edge cases that might return incomplete results.
- `timeseries_query` is a complicated abstraction that includes quite a bit that is unnecessary for my use case. It also is a few different types of data transformations which i would like to avoid.

If there are any harder concerns, i'd be happy to discuss them more in depth